test/models/lockable_test.rb

```diff
@@ -37,6 +37,17 @@ def setup
     end
   end
 
+  test "should read failed_attempts from database when incrementing" do
+    user = create_user
+    initial_failed_attempts = user.failed_attempts
+    same_user = User.find(user.id)
+
+    user.valid_for_authentication?{ false }
```

Comment:
Just out of curiosity, what's the benefit of calling `#valid_for_authentication?` instead of `#user.increment_failed_attempts`?

Reply:
Afaik `increment_failed_attempts` is not available in the `3-stable` branch.
The incrementation is actually done in the [`valid_for_authentication?` method](https://github.com/plataformatec/devise/blob/3-stable/lib/devise/models/lockable.rb#L103)

Reply:
Oh, that's correct. That's my bad, sorry 😞